RecoLocalCalo/HcalRecAlgos/src/PulseShapeFitOOTPileupCorrection.cc

```diff
@@ -304,7 +301,11 @@ void PulseShapeFitOOTPileupCorrection::phase1Apply(const HBHEChannelInfo& channe
   psfPtr_->setinvertpedSig2(1./(averagePedSig2GeV));
 
   if(channelData.hasTimeInfo()) { 
-    ts4Max_=vts4Max_[1]; ts4Chi2_=vts4Chi2_[1]; 
+    ts4Chi2_=vts4Chi2_[1];
+    if(channelData.id().depth()==1) ts4Max_=vts4Max_[1];
+    if(channelData.id().depth()>1) ts4Max_=vts4Max_[2];
```

Comment:
```else if (channelData.id().depth()>1) ts4Max_=vts4Max_[2];```

In any case: negative depth's are not possible (correct?): in that case, you should simply write
```else ts4Max_=vts4Max_[2];```

Otherwise, ```ts4Max``` could in principle remain undefined, and a default value should be supplied